File: app/src/main/java/abandonedstudio/app/cookingapp/Database/PreparationStep.java
package abandonedstudio.app.cookingapp.Database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "preparation_steps", foreignKeys = @ForeignKey(entity = Dish.class, parentColumns = "dishId", childColumns = "dish_id", onDelete = ForeignKey.CASCADE, onUpdate = ForeignKey.CASCADE))
public class PreparationStep {

    public PreparationStep(int stepNumber, String stepDescription, int dishId) {
        this.stepNumber = stepNumber;
        this.stepDescription = stepDescription;
        this.dishId = dishId;
    }

    @PrimaryKey(autoGenerate = true)
    private int stepId;

    @ColumnInfo(name = "step_number")
    private int stepNumber;

    @ColumnInfo(name = "step_description")
    private String stepDescription;

    @ColumnInfo(name = "dish_id")
    private int dishId;

    public int getStepId() {
        return stepId;
    }

    public void setStepId(int stepId) {
        this.stepId = stepId;
    }

    public int getStepNumber() {
        return stepNumber;
    }

    public String getStepDescription() {
        return stepDescription;
    }

    public int getDishId() {
        return dishId;
    }
}

File: app/src/main/java/abandonedstudio/app/cookingapp/core/data/local/room/preparation_step/PreparationStep.kt
package abandonedstudio.app.cookingapp.core.data.local.room.preparation_step

import abandonedstudio.app.cookingapp.core.data.local.room.dish.Dish
import androidx.room.*

@Entity(tableName = "preparation_steps", indices = [Index("dish_id")], foreignKeys = [ForeignKey(entity = Dish::class, parentColumns = ["dishId"], childColumns = ["dish_id"], onDelete = ForeignKey.CASCADE, onUpdate = ForeignKey.CASCADE)])
data class PreparationStep(

    @ColumnInfo(name = "step_number")
    var stepNumber: Int,

    @ColumnInfo(name = "step_description")
    var stepDescription: String,

    @ColumnInfo(name = "dish_id")
    var dishId: Int,

    @PrimaryKey(autoGenerate = true)
    var stepId: Int? = null
)
